hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/procedures/ValidateAuditLockProcedure.scala

```diff
@@ -0,0 +1,307 @@
+
+            // Generate result
+            val (result, issuesFound, details) = formatValidationResults(validationResults)
+
+            Seq(Row(displayName, result, windows.size, issuesFound, details))
+          }
+        }
+      }
+    } catch {
+      case e: Exception =>
+        val errorMessage = s"Validation failed: ${e.getMessage}"
+        Seq(Row(displayName, "ERROR", 0, -1, errorMessage))
+    }
+  }
+
+  /**
+   * Parses an audit file and extracts transaction window information.
+   */
+  private def parseAuditFile(pathInfo: StoragePathInfo, storage: org.apache.hudi.storage.HoodieStorage): Option[TransactionWindow] = {
+    val filename = pathInfo.getPath.getName
+
+    Try {
+      // Read file content using Hudi storage API
+      val inputStream = storage.open(pathInfo.getPath)
+      val content = try {
+        scala.io.Source.fromInputStream(inputStream).mkString
+      } finally {
+        inputStream.close()
+      }
+
+      // Parse JSONL content
+      val lines = content.split('\n').filter(_.trim.nonEmpty)
+      val jsonObjects = lines.map(OBJECT_MAPPER.readTree)
+
+      if (jsonObjects.isEmpty) {
+        None
+      } else {
+
+        // Extract transaction metadata
+        val firstObject = jsonObjects.head
+        val ownerId = firstObject.get("ownerId").asText()
+        val transactionStartTime = firstObject.get("transactionStartTime").asLong()
+
+      // Find first START timestamp
+      val startRecords = jsonObjects.filter(_.get("state").asText() == "START")
+      val startTimestamp = if (startRecords.nonEmpty) {
+        startRecords.head.get("timestamp").asLong()
+        } else {
+          transactionStartTime // fallback to transaction start time
+        }
+
+        // Find last END timestamp
+        val endRecords = jsonObjects.filter(_.get("state").asText() == "END")
+        val endTimestamp = if (endRecords.nonEmpty) {
+          Some(endRecords.last.get("timestamp").asLong())
+        } else {
+          None
+        }
+
+        // Find last expiration time as fallback
+        val lastExpirationTime = if (jsonObjects.nonEmpty) {
+          val lastObject = jsonObjects.last
+          if (lastObject.has("lockExpiration")) {
+            Some(lastObject.get("lockExpiration").asLong())
+          } else {
+            None
+          }
+        } else {
+          None
+        }
+
+        Some(TransactionWindow(
+        ownerId = ownerId,
+        transactionStartTime = transactionStartTime,
+        startTimestamp = startTimestamp,
+        endTimestamp = endTimestamp,
+        lastExpirationTime = lastExpirationTime,
+          filename = filename
+        ))
+      }
+    } match {
+      case Success(window) => window
+      case Failure(_) => None // Skip corrupted files
+    }
+  }
+
+  /**
+   * Validates transaction windows for overlaps and proper closure.
+   */
+  private def validateTransactionWindows(windows: Seq[TransactionWindow]): ValidationResults = {
```

Comment:
Similar here, reuse part of the logic between `LockAuditingCommand` and `ValidateAuditLockProcedure`